doc/chips/10.rst

```diff
@@ -460,50 +498,17 @@ their type when copies of it must be made, such as when the type is passed to a
 implement reference counting for their type.  For these situations, the type
-designer may define a copy initializer, to handle the final state of the type's
-fields once a shallow bit copy has been made.
-
-Note that the cases handled by a copy initializer are not the same as user
-specified calls of the form:
+designer may define a copy initializer, to handle the state of the type's
+fields.  The copy initializer would be an initializer with a single argument, of
+the same type as the type being created.  The copy initializer would be called
+both in these cases and in user specified calls of the form:
 
 .. code-block:: chapel
 
    var x = new R(y);
 
-Such statements would be handled by an initializer with a single argument of the
-same type as the object being created, if the type designer intends for such
-actions to occur.
-
-Copy initializers will be quite different from normal initializers.  In the
-first place, the object in question is expected to be “initialized” once the
-body is entered – the shallow bit copy will have provided all fields with an
-initial value, and the copy initializer is intended to mend or improve upon what
-has been done.  Thus, a division into phases similar to those in normal
-initializers is unnecessary in the copy initializer body.  The body will behave
-in a similar manner to Phase 2 of the normal initializers, with the ability to
-modify a field in any order and to call methods on the instance as a whole.
-However, since the type designer would otherwise be unable to directly control
-the value of a ``const`` field, ``const`` fields can be modified in the body of
-a copy initializer.  Due to concerns about providing a field value to a method
-which could change unexpectedly in the middle of the method's operations,
-ideally ``const`` fields would only be modifiable before the first method call
-on the instance during the copy initializer body – this may prove infeasible,
-though.  If doable, this permissive behavior could be extended to Phase 2 of the
-normal initializer body.
-
-The method header for a copy initializer will not allow as much variation as a
-normal initializer – the compiler will be the only client directly making calls
-to the method and will thus expect a single style of header.  This trivially
-means that a type cannot define multiple copy initializers, as doing so would
-result in ambiguity.
-
-Due to the unfinished status of record method inheritance, no support for calls
-to a parent copy initializer will be provided; any necessary operations on a
-parent field must be done in the body of the child copy initializer.  When
-record method inheritance is finalized, this policy may change.
-
-Final details for the syntax of this particular style of initializer have not
-been fully ironed out, stay tuned!
+A more complicated strategy, similar to that employed in D's postblit, may be
+considered at a later time, but for now, we view this support as sufficient.
```

Comment:
At this point I think you should refer to the most excellent CHIP 13: https://github.com/chapel-lang/chapel/blob/master/doc/chips/13.rst which describes in detail when the compiler will call a copy initializer.